lib/traject/solr_json_writer.rb

```diff
@@ -162,11 +167,12 @@ def send_single(c)
         msg = "Solr error response: #{resp.status}: #{resp.body}"
       end
       logger.error "Could not add record #{c.record_inspect}: #{msg}"
-      logger.debug(c.source_record.to_s)
+      logger.debug("\t" + exception.backtrace.join("\n\t")) if exception
```

Comment:
I wonder if we can remove this change now, since the `FATAL` error will get logged. The scenario I'm thinking about is a large indexing job (500,000 records) with multiple large failures may fill up the logs rather quickly if max_skipped = -1. What do you think?

Reply:
yeah good point. I think this is maybe why I wasn't logging stack traces so much first time. Under _normal_ use, I don't want to overwhelm people with info, but that's differnet than "trying to figure out what is going wrong" debugging use. And is part of why I'm sending em to DEBUG now -- I imagine you'll generally turn off DEBUG in that scenario. 

BUT. Wait. So, this is confusing. Here is what I was thinking. By "the FATAL error", do you mean the one in CommandLine? 

Cause... This exception that gets logged here never actually makes it to CommandLine, since it's a different MaxSkippedRecordsExceeded error that gets raised out on max_skipped exceeded. Which has lost the original stack trace. :(

So what happens here is _every_ time a record can't be added, it's failure gets logged (as before), and stack trace to debug (new).  Then only when max skipped is exceeded does a MaxSkippedRecordsExceeded get raised, and reported/logged higher up in stack in CommandLine (under normal use) -- but at that point you no longer know the (possibly _several different_) original stack traces that led to MaxSkippedRecordsExceeded, right?

But this is basically for your use cases, if YOU don't think you need it, I'll take it out! Want to play around with it and see? Or not?

Reply:
Ah yeah I see what you mean now.

I'd say leave it in then. 👍

Reply:
Hmm, if @max_skipped is 0, we could just re-raise the real cause exception, instead of a MaxSkippedRecordsExceeded. 

But if it's more than 0, we don't know what it is, there could be several. 

And if it's -1... and we _don't_ log the stack trace each time... then you'd definitely _never_ get the stack trace. I thought getting the stack trace was something you wanted, which is why I added it!  if you don't want it, I'll take it out again. :)

Reply:
Yeah I think we do want it. So I'm 👍  to leaving it in. (negating my previous comment) :)